internal/app/service.go

```diff
@@ -120,7 +119,8 @@ func (svc *Service) AddRoute(route string, handler func(nethttp.ResponseWriter,
 // AddBackgroundPublisher will create a channel of provided capacity to be
 // consumed by the MessageBus output and return a publisher that writes to it
 func (svc *Service) AddBackgroundPublisher(capacity int) interfaces.BackgroundPublisher {
-	bgchan, pub := newBackgroundPublisher(capacity)
+	//TODO: error if not message bus?
+	bgchan, pub := newBackgroundPublisher(svc.config.Trigger.EdgexMessageBus.PublishHost.PublishTopic, capacity)
```

Comment:
I see now, still using `PublishTopic` from config, but don't have access to it when needing to format the topic. Would it be useful to have two factory methods with one allowing the topic to be overridden and this one which uses the default configured topic?